src/Components/Psychologists/index.js

```diff
@@ -1,9 +1,250 @@
 import styles from './psychologists.module.css';
+import { useState, useEffect } from 'react';
+import List from './List';
+import Form from './Form';
+import Modal from './Modal';
+import AvailabilityTable from './AvailabilityTable';
+import { PSYCHOLOGIST_FORM, PSYCHOLOGIST_AVAILABILITY } from './utils/psychologist-inputs-utils';
+
+const itemOnEditInitialState = {
+  address: '',
+  availability: {
+    monday: {
+      availability: false,
+      from: 1200,
+      to: 1200
+    },
+    tuesday: {
+      availability: false,
+      from: 1200,
+      to: 1200
+    },
+    wednesday: {
+      availability: false,
+      from: 1200,
+      to: 1200
+    },
+    thursday: {
+      availability: false,
+      from: 1200,
+      to: 1200
+    },
+    friday: {
+      availability: false,
+      from: 1200,
+      to: 1200
+    },
+    sunday: {
+      availability: false,
+      from: 1200,
+      to: 1200
+    },
+    saturday: {
+      availability: false,
+      from: 1200,
+      to: 1200
+    }
+  },
+  email: '',
+  firstName: '',
+  lastName: '',
+  password: '',
+  phone: '',
+  username: ''
+};
 
 function Psychologists() {
+  const [psychologists, setPsychologist] = useState([]);
+  const [toggleForm, setToggleForm] = useState(false);
+  const [itemOnEdit, setItemOnEdit] = useState(itemOnEditInitialState);
+  const [isEditing, setIsEditing] = useState(false);
+  const [showAvailabilityModal, setShowAvailabilityModal] = useState(false);
+  const [showConfirmModal, setShowConfirmModal] = useState(false);
+  const [psychologistAvailability, setPsychologistAvailability] = useState({});
+
+  useEffect(() => {
+    getPsychologists();
+    console.log(psychologists);
+  }, []);
+
+  const getPsychologists = () => {
+    fetch(`${process.env.REACT_APP_API}/psychologists`)
+      .then((response) => response.json())
+      .then((response) => {
+        setPsychologist(response.data);
+      })
+      .catch((error) => error);
+  };
+
+  const toggleFormDisplay = () => {
+    setToggleForm(!toggleForm);
+    setItemOnEdit(itemOnEditInitialState);
+  };
+
+  const toggleAvailabilityModal = (e, psychologist) => {
+    if (!showAvailabilityModal) {
+      e.stopPropagation();
+      setPsychologistAvailability(psychologist.availability);
+      setShowAvailabilityModal(!showAvailabilityModal);
+    } else {
+      setShowAvailabilityModal(!showAvailabilityModal);
+    }
+  };
+
+  const toggleConfirmModal = (e, psychologist) => {
+    if (!showConfirmModal) {
+      e.stopPropagation();
+      setItemOnEdit(psychologist);
+      setShowConfirmModal(true);
+    } else {
+      setItemOnEdit(itemOnEditInitialState);
+      setShowConfirmModal(false);
+    }
+  };
+
+  const handleEdit = (psychologist) => {
+    toggleFormDisplay();
+    setItemOnEdit(psychologist);
+    setIsEditing(true);
+  };
+
+  const onChange = (e) => {
+    if (e.target.name.match(/^(firstName|lastName|username|password|email|address)$/)) {
```

Comment:
very complex, use different states for each input. You should keep in mind this function should be understandable to the rest of the team